onadata/auth.py

```diff
@@ -28,3 +28,24 @@ def authenticate(self, request):
         user = authenticate(remote_user=request.META.get(self.header))
         if user and user.is_active:
             return (user, None)
+
+
+class QedRemoteUserAttributeMiddleware(object):
+    username_header = 'HTTP_X_AUTH_USERNAME'
+    email_header = 'HTTP_X_AUTH_EMAIL'
+
+    def process_request(self, request):
+        username = request.META.get(self.username_header, None)
+        email = request.META.get(self.email_header, None)
+
+        if hasattr(request, 'user') and request.user.is_authenticated and \
+                username and email:
```

Comment:
Is there any reason for checking whether `HTTP_X_AUTH_USERNAME` is non-empty?

Reply:
I don't want to spend time envisioning all strange misconfigurations and what the impact might be, I prefer to only do something when I know that I'm in a correct state.